### DISCUSSION TITLE
Hourly, daily, weekly and monthly tasks in 1 DAG

### DISCUSSION THREAD

**Discussion by MF-96** [apache/airflow#22680] [Q&A]
Hi,
I am considering migartion to Airflow and and wondering how I can build this pipeline: 
1. 2x tasks running on hourly schedule (getting data from other DB)
2. 2x tasks running on daily schedule (transforming data collected by tasks from point 1),
3. 1x task running on weekly schedule (creating Excel report based on tasks from point 2)
4. 1x task running on monthly schedule (creating Excel report based on tasks from point 2)

General outline of pipeline:
![Diagram bez tytułu drawio - diagrams net](https://user-images.githubusercontent.com/96316666/161229573-89358e31-a65e-4e02-85fb-41c28e5f28ad.jpg)

Question: is it possible to create this pipeline in 1 DAG or I have to create x DAGs because of the different shedulers?

**Comment by potiuk** [comment]
You need to have separate DAGS. In the near future you will be able to make "logical" connection between them, following the https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-48+Data+Dependency+Management+and+Data+Driven+Scheduling but until then, the relation between them is mostly in the data they would use.

You could "theorethically" skip some of the parts of hourly DAG and run them "really" on a weekly or monthly base, but that is far more hassle than it works - having separate dags also gives you more control over re-running parts of the pipelines, backfilling and the like.

**Reply by MF-96** [reply]
Thanks for fast reply!
If I can, I would also like to ask if there is a possibility to make dependency between daily dag and hourly dag, so daily dag only triggers when every hourly dag from previous day completed succssefully ?

**Reply by potiuk** [reply]
no

**Comment by terrynice** [comment]
anything new about this topic?

**Reply by potiuk** [reply]
NO.

**Comment by potiuk** [comment]
If you want similar dags at different schedules you can always use Dynamic Dags https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#dynamic-dags. and use the same code to generate different dags at different schedule
